exercises/csharp/TriviaGame/TriviaGame/Game.cs

```diff
@@ -19,92 +19,111 @@ public class Game
         LinkedList<string> sportsQuestions = new LinkedList<string>();
         LinkedList<string> rockQuestions = new LinkedList<string>();
 
-        int currentPlayer = 0;
-        bool isGettingOutOfPenaltyBox;
+        int _currentPlayer = 0;
+        bool _leavingPenaltyBox;
 
         public Game()
         {
-            for (int i = 0; i < 50; i++)
+            for (int i = 0; i < CardQuestionNumber; i++)
             {
                 popQuestions.AddLast("Pop Question " + i);
                 scienceQuestions.AddLast(("Science Question " + i));
                 sportsQuestions.AddLast(("Sports Question " + i));
-                rockQuestions.AddLast(CreateRockQuestion(i));
+                rockQuestions.AddLast(("Rock Question " + i));
             }
         }
 
-        public string CreateRockQuestion(int index)
-        {
-            return "Rock Question " + index;
-        }
+        //public string CreateRockQuestion(int index)
+        //{
+        //    return "Rock Question " + index;
+        //}
 
-        public bool IsPlayable()
-        {
-            return (HowManyPlayers() >= 2);
-        }
+        // Unused function - commented out
 
-        public bool Add(string playerName)
+        //public bool IsPlayable()
+        //{
+        //    return (HowManyPlayers() >= 2);
+        //}
+
+        public bool AddPlayer(string playerName)
         {
 
 
             players.Add(playerName);
-            places[HowManyPlayers()] = 0;
-            purses[HowManyPlayers()] = 0;
-            inPenaltyBox[HowManyPlayers()] = false;
+            places[TotalNumberOfPlayers()] = 0;
+            playerGoldCoins[TotalNumberOfPlayers()] = 0;
+            inPenaltyBox[TotalNumberOfPlayers()] = false;
 
             Console.WriteLine(playerName + " was added");
             Console.WriteLine("They are player number " + players.Count);
             return true;
         }
 
-        public int HowManyPlayers()
+        public int TotalNumberOfPlayers()
         {
             return players.Count;
         }
 
-        public void Roll(int roll)
+        public void NextPlayerRolls(int roll)
         {
-            Console.WriteLine(players[currentPlayer] + " is the current player");
+            Console.WriteLine(players[_currentPlayer] + " is the current player");
             Console.WriteLine("They have rolled a " + roll);
 
-            if (inPenaltyBox[currentPlayer])
+            void LeftPenaltyBox()
```

Comment:
You're using local functions here which is cool. Just be aware that some people don't like local functions/haven't used them before. They're a relatively new c# feature